src/components/mapComponents/maps/mapWithPopup/index.tsx

```diff
@@ -189,6 +217,7 @@ const MapWithPopup: React.FC<MapWithPopupProps> = ({
           onChange={(event) => setSearchInput(event.target.value)}
         />
       </div>
+      {toggleViewCard && createPortal(toggleViewCard, toggleViewDiv.current)}
```

Comment:
A bit of explanation for this:

I wanted to make the toggle view card a [google maps custom control](https://developers.google.com/maps/documentation/javascript/examples/control-custom) so it's automatically positioned nicely as the window is resized (the styling for the given component is relative to the top right corner of the map). However, custom controls must be native DOM elements, not React components, We can attach a dummy div as our control, then use a [react portal](https://reactjs.org/docs/portals.html) to attach a component. 

Unfortunately, since we render the map in a `useEffect` hook, if we create a plain div through `document.createElement`, I believe what happens is the div with the component gets overwritten by an empty div without the component during a re-render, so nothing shows up - we can use a `useRef` hook for a "stateful" div and properly show the card.

tl;dr i don't like useEffect